docs/rule/no-builtin-form-components.md

```diff
@@ -0,0 +1,65 @@
+# no-builtin-form-components
+
+Ember's built-in form components use two-way data binding, where the property as `@value` or `@checked` is mutated by user interaction. This goes against the Data Down Actions Up principle, goes against Glimmer Components’ intention to have immutable arguments, and is [discouraged by the Ember Core team](https://www.pzuraq.com/on-mut-and-2-way-binding/).
```

Comment:
The pzuraq post is fairly old and mostly concerned with the `mut` helper. Would be nice to link to a better resource here.